src/Core/Domain/Product/ValueObject/Visibility.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+
+/**
+ * Defines where product should be visible.
+ */
+final class Visibility
+{
+    /**
+     * @var string defines that product is visible everywhere - e.g category, product, search pages
+     */
+    public const EVERYWHERE = 'both';
+
+    /**
+     * @var string - defines that the product is visible only in catalog page.
+     */
+    public const CATALOG = 'catalog';
+
+    /**
+     * @var string - defines that the product is visible only in search results.
+     */
+    public const SEARCH = 'search';
+
+    /**
+     * @var string - defines that product should not appear anywhere.
+     */
+    public const NOWHERE = 'none';
+
+    public const AVAILABLE_VISIBILITY = [
+        self::EVERYWHERE,
+        self::CATALOG,
+        self::SEARCH,
+        self::NOWHERE,
+    ];
+
+    /**
+     * @var string
+     */
+    private $visibility;
+
+    /**
+     * @var bool
+     */
+    private $isAvailableForOrder;
+
+    /**
+     * @var bool
+     */
+    private $isWebOnly;
```

Comment:
Are `$isAvailableForOrder`  and `$isWebOnly` really part of `Visibility` VO?

Reply:
they are related  - how do you suggest to move these options since we will need to identify it anyway

Reply:
It's just an attribute of Product. In this case, I don't think they are part of `Visibility` VO.

Reply:
same problem - do you suggest to set them separately in the command?

Reply:
Yeah.